### PR TITLE
Integrated AR system CBM

### PR DESCRIPTION
**Pull request by Dan1ss1mo**
#### Summary
Content "Integrated AR system CBM"

#### Purpose of change
Adding new cbm what would give you all the benefits of having AR glasses.

#### Describe alternatives you've considered
Maybe some balance tweaks

#### Testing
Successfully spawned and intalled cbm. Everything works as intended 

#### Additional context
You just can't be a cool cyborg without cool overlay hud

### CODE REVIEW

**Comment by Dan1ss1mo** [comment]
> I dunno your vision on this, but having "ZOOM" would imply that this intra-eye implant is able to view as far as the telescopic lenses, without... having telescopic lenses.

through augmented reality i guess? Tho I'm fine with removing it if it is too out of place for this cbm

File: data/json/items/bionics.json [hunk 1]
```diff
@@ -70,6 +70,17 @@
     "difficulty": 1,
     "installation_data": "AID_bio_alarm"
   },
+  {
+    "id": "bio_ar",
+    "copy-from": "bionic_general_npc_usable",
+    "type": "BIONIC_ITEM",
+    "name": { "str": "Integrated AR System CBM" },
+    "looks_like": "bio_int_enhancer",
+    "description": "This bionic module integrate augmented reality interface into user's eyes.",
```
Comment (Night-Pryanik) [inline data/json/items/bionics.json:79]:
```suggestion
    "description": "This bionic module integrates augmented reality interface into user's eyes.",
```

File: data/json/items/fake.json [hunk 2]
```diff
@@ -38,6 +38,37 @@
     "use_action": [ "DIG" ],
     "qualities": [ [ "DIG", 2 ] ]
   },
+  {
+    "id": "integrated_ar",
+    "type": "TOOL_ARMOR",
+    "copy-from": "fake_item",
+    "name": { "str": "Integrated AR System" },
+    "description": "Provide user with augmented reality overlay allowing the user to zoom, and it displays handy information about local topology.",
```
Comment (Night-Pryanik) [inline data/json/items/fake.json:46]:
```suggestion
    "description": "Provides user with augmented reality overlay allowing the user to zoom, and it displays handy information about local topology.",
```

File: data/json/bionics.json [hunk 3]
```diff
@@ -41,6 +41,15 @@
     "mutation_conflicts": [ "LEG_TENTACLES", "LEG_TENT_BRACE", "GASTROPOD_FOOT", "GASTROPOD_BALANCE", "AMORPHOUS" ],
     "flags": [ "BIONIC_FAULTY" ]
   },
+  {
+    "id": "bio_ar",
+    "type": "bionic",
+    "name": { "str": "Integrated AR System" },
+    "description": "Provide user with augmented reality overlay allowing the user to zoom, and it displays handy information about local topology.",
+    "occupied_bodyparts": [ [ "eyes", 1 ] ],
+    "flags": [ "USES_BIONIC_POWER" ],
```
Comment (Maleclypse) [inline data/json/bionics.json:50]: You are missing `"mutation_conflicts"` for all the eye mutations.

File: data/json/npcs/exodii/exodii_merchant_itemlist.json [hunk 4]
```diff
@@ -177,7 +177,8 @@
       [ "bio_scent_mask", 10 ],
       [ "bio_jointservo", 10 ],
       [ "bio_taste_blocker", 10 ],
-      [ "bio_trickle", 10 ]
+      [ "bio_trickle", 10 ],
+      [ "bio_ar", 10 ]
```
Comment (Maleclypse) [inline data/json/npcs/exodii/exodii_merchant_itemlist.json:181]: You don't have to do so but I'd appreciate if you added these to the renfield CBM dissects in Xedra Evolved and the scavenger/pirate CBM dissect lists in Aftershock.

File: data/json/bionics.json (hunk 3 above)
Comment (Dan1ss1mo) [reply data/json/bionics.json:50]: Should it tho? I can see it conflicting with eyestalks, but somthing like lizard eyes seems fine

Comment (Dan1ss1mo) [reply data/json/bionics.json:50]: I think I'm going to flavour it like lenses that go over you're eyes. I'll tweak the description to reflect that and set the list of conflict similar to protective lenses

**Comment by BroseidonMOO** [comment]
Hopefully this comment is appropriate, please let me know if it is not. The test failure appears unrelated to the PR, and is probably worth rerunning. Thank you!

**Comment by Dan1ss1mo** [comment]
> Hopefully this comment is appropriate, please let me know if it is not. The test failure appears unrelated to the PR, and is probably worth rerunning. Thank you!

Thank you as well) The failed test shouldn't be that big of a deal, as only required ones matter for the most part, atleast for json pr's

**Comment by Maddremor** [comment]
While I approve of all of the individual functionalities implemented here, I think it would be more interesting to split it up over multiple CBMs. That way the player can find a cool new gadget multiple times instead of having a one-and-done experience.

**Comment by Dan1ss1mo** [comment]
> While I approve of all of the individual functionalities implemented here, I think it would be more interesting to split it up over multiple CBMs. That way the player can find a cool new gadget multiple times instead of having a one-and-done experience.

Imo, splitting it would make it not very cool. Each time i find stuff like cranial flashlight or finger lighter i feel like i got scammed. And i definitely can't be bothered to install something like that, even with Rubik. Having a set of nice utilities in cbm on the other hand seems pretty neat to me. Non of them are irreplaceable, but having them all in a single package always at hand is the selling point for me

**Comment by Dan1ss1mo** [comment]
I'm fine with making in rarer\take up more slots tho

**Comment by Maleclypse** [comment]
> I'm fine with making in rarer\take up more slots tho

You could also make it an upgradeable cbm and split it up that way.

**Comment by Dan1ss1mo** [comment]
> > I'm fine with making in rarer\take up more slots tho
> 
> You could also make it an upgradeable cbm and split it up that way.

That's an intresting idea, tho I'm not sure how to implement that. We don't have any upgradeable cbms currently, right? What do you think would be a good way to handle that?

**Pull request merged**